### PR TITLE
[LSM] Misc changes after local tests

### PR DESCRIPTION
**Pull request by sampocs**

### CODE REVIEW

File: x/interchainquery/keeper/keeper.go [hunk 1]
```diff
@@ -78,6 +78,9 @@ func (k *Keeper) RetryICQRequest(ctx sdk.Context, query types.Query) error {
 		return err
 	}
 
+	// TODO [LSM]: Investigate why we need to force a unique ID when retrying queries
```
Comment (riley-stride) [inline x/interchainquery/keeper/keeper.go:81]: Could you expand on this issue?

File: x/stakeibc/keeper/icqcallbacks_validator_exchange_rate.go [hunk 2]
```diff
@@ -137,6 +137,9 @@ func ValidatorExchangeRateCallback(k Keeper, ctx sdk.Context, args []byte, query
 
 		EmitFailedLSMLiquidStakeEvent(ctx, hostZone, lsmTokenDeposit, errorMessage)
 		k.Logger(ctx).Error(errorMessage)
+
```
Comment (riley-stride) [inline x/stakeibc/keeper/icqcallbacks_validator_exchange_rate.go:140]: piconit: can you clarify the comment on ln 132? "event rejects tx" is unclear to me

**Review by riley-stride — APPROVED** [review]
Changes lgtm. Are you planning to turn these scripts into full integration tests?

File: x/interchainquery/keeper/keeper.go (hunk 1 above)
Comment (sampocs) [reply x/interchainquery/keeper/keeper.go:81]: For some reason, it seemed like the query events weren't getting emitted when I tried to resubmit with the same query ID. I'm going to reach out to Andrew - I assume there's an assumption the relayer makes

File: x/records/keeper/transfer.go [hunk 3]
```diff
@@ -67,9 +67,6 @@ func (k Keeper) IBCTransferLSMToken(
 	hostZoneDepositAddress string,
 	hostZoneDelegationICAAddress string,
 ) error {
-	// Update deposit status
```
[the remark this answers is not in the document]
Comment (sampocs) [reply x/records/keeper/transfer.go:70]: updated earlier! it happens already in FinishLSMLiquidStake, this was a dupe

**Comment by sampocs** [comment]
> Changes lgtm. Are you planning to turn these scripts into full integration tests?

Yeah most definitely

**Pull request merged**